src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -18,13 +18,20 @@ export interface CapitalCostConfig {
   cutoff: string;
   decimals: number;
 }
+type ChainIdAsString = string;
+interface CapitalCostConfigOverride {
+  default: CapitalCostConfig;
+  routeOverrides?: Record<ChainIdAsString, Record<ChainIdAsString, CapitalCostConfig>>;
+}
 export interface RelayFeeCalculatorConfig {
   nativeTokenDecimals?: number;
   gasDiscountPercent?: number;
   capitalDiscountPercent?: number;
   feeLimitPercent?: number;
   capitalCostsPercent?: number;
-  capitalCostsConfig?: { [token: string]: CapitalCostConfig };
+  capitalCostsConfig?: {
+    [token: string]: CapitalCostConfig | CapitalCostConfigOverride;
+  };
```

Comment:
personally would have made another property here called `capitalCostsConfigOverride` 
 which is the same shape as `capitalCostsConfig` and prioritized using that before falling back to `capitalCostsConfig`. i think this would have simplified some of the code as you wouldnt need special checks, just a fallback if a value didnt exist.

Reply:
I'm not strongly opinionated, as this is pretty tricky. If we didn't have to worry about backcompat, my preference would be a config that looks like this:
```ts
export interface RelayFeeCalculatorConfig {
  ...
  defaultCapitalCostsPercent: number;
  tokenCapitalCosts?: { [token: string]: number };
  // Note: routes are encoded as strings 1-10 would be the route from chain ID 1 to chain ID 10.
  routeCapitalCosts?: { [token: string]: { [route: string]: number } };
}
```

In that world, you could just store these and do a series of fallbacks that's pretty easy to understand:
```ts
const routeCost = routeCapitalCosts?.[token]?.[routeString];
if (routeCost) return routeCost;
const tokenCost = tokenCapitalCosts?.[token];
if (tokenCost) return tokenCost;
return defaultCost;
```

I think this strategy would get rid of a lot of the pre-parsing and runtime checks that we are doing in this PR and reduce the complexity of the nesting, at the cost of having more config variables.

However, one problem is that we _may_ have to deal with capitalCostsConfig for backwards compatibility reasons. Although, we could publish a new breaking change where we immediately drop this config variable. Then in other applications, when we update to that version, we have to update our config passing simultaneously.

Thoughts? Personally, I think the existing approach is reasonable, just wanted to mention a possible alternative (that may or may not be better).

Reply:
@mrice32 @james-a-morris yes perfectly reasonable, i was probably missing some context, but either approach seems fine to me

Reply:
Agreed, either way seems okay. I would personally prefer less tech debt and complexity over backwards compatibility since we have explicit releases to protect end-users.

Ultimately up to you @james-a-morris!